Model/PayolutionInvoicePaymentMethod.php

```diff
@@ -175,6 +194,25 @@ public function getMetaDataFieldNames()
             'shipping_countries',
             'billing_countries',
             'billing_shipping',
+            'trusted_shop',
+            'payolution_terms_url',
         ];
     }
+
+    /**
+     * @throws InputException
+     *
+     * @since 1.2.0
+     */
+    public function onBeforeTransactionCreation()
+    {
+        parent::onBeforeTransactionCreation();
+
+        $oRequest = Registry::getRequest();
+
+        if ($this->_oPayment->oxpayments__trusted_shop->value &&
```

Comment:
Please encapsulate this in a method called _isTermsAccepted()

Reply:
👍